src/components/Dialog/CloneDialog/CloneDialog.js

```diff
@@ -129,30 +138,50 @@ export class CloneDialog extends React.Component {
   onErrorDismissed = () => this.setState({ error: null });
 
   render() {
+    const {
+      LoadingComponent,
+      vm,
+      virtualMachines,
+      namespaces,
+      persistentVolumeClaims,
+      dataVolumes,
+      requestsDatavolumes,
+      requestsPVCs,
+    } = this.props;
+
     const formFields = getFormFields(
-      this.props.namespaces,
-      this.props.vm,
-      this.props.persistentVolumeClaims,
-      this.props.dataVolumes,
-      this.props.virtualMachines
+      getLoadedData(namespaces, []),
+      vm,
+      getLoadedData(persistentVolumeClaims, []),
+      getLoadedData(dataVolumes, []),
+      getLoadedData(virtualMachines, [])
     );
-    const { LoadingComponent } = this.props;
+
+    const dataVolumesValid = requestsDatavolumes ? dataVolumes && dataVolumes.loaded && !dataVolumes.loadError : true;
+    const pvcsValid = requestsPVCs
+      ? persistentVolumeClaims && persistentVolumeClaims.loaded && !persistentVolumeClaims.loadError
+      : true;
+
     const footer = this.state.cloning ? (
       <LoadingComponent />
     ) : (
       <React.Fragment>
-        <Button bsStyle="default" className="btn-cancel" onClick={this.props.onClose}>
+        <Button bsStyle="default" className="btn-cancel" onClick={this.props.cancel}>
           Cancel
         </Button>
-        <Button bsStyle="primary" onClick={this.cloneVm} disabled={!this.state.valid}>
+        <Button
+          bsStyle="primary"
+          onClick={this.cloneVm}
+          disabled={!(this.state.valid && dataVolumesValid && pvcsValid)}
```

Comment:
We should handle error state more explicitly - the user should be informed about transition from `loading` to `error` state.
I believe rendering info about just the most serious issue (as filtered by `Firehose`) is enough.